openmls/src/framing/test_framing.rs

```diff
@@ -411,12 +413,13 @@ fn unknown_sender(ciphersuite: &'static Ciphersuite, backend: &impl OpenMlsCrypt
         .create_commit(params, backend)
         .expect("Error creating Commit");
 
-    let staged_commit = group_alice
-        .stage_commit(&create_commit_result.commit, &proposal_store, &[], backend)
-        .expect("Could not stage Commit");
     group_alice
-        .merge_commit(staged_commit)
-        .expect("error merging commit");
+        .merge_staged_commit(
```

Comment:
Why not use `.merge_commit()` here? Just flagging it here, but it happens in other places as well.

Reply:
Good point. This indeed happens in pretty much all tests now. But now I'm confused why both calls exist, since `merge_commit` also takes a `StagedCommit` as input. Also, `merge_commit` seems to be `test`/`test-utils` only, so it's probably better to use `merge_staged_commit`, so we use code that's actually used by the `MlsGroup` API as opposed to test-only code.

Reply:
I agree that this is in a messy state right now. The reason why `merge_staged_commit` existed was that `CoreGroup` was public. I'm thinking that this should get cleaned up as part of #645. We'll probably end up having just one function that replaces `merge_staged_commit`, `merge_commit` and `merge_commit_take_message_secrets` and that takes a `StagedCommit` as its only parameter. So in that sense, it doesn't really matter which one you pick, but you picked the one that introduces the biggest change in parameters.

Reply:
Fair enough. If we're going to "merge" the three functions and end up with only one that takes a `StagedCommit` as input, it probably makes sense that I change everything over to `merge_commit` right now, while I'm touching the code in all of these places anyway.